docs/avro.rst

```diff
@@ -1,98 +1,47 @@
-some sort of schemas. These libraries include `Avro <http://avro.apache.org>`_,
-`Thrift <http://thrift.apache.org>`_, and `Protocol Buffers <https://github.com/google/protobuf>`_.  The advantage of
-having a schema is that it clearly specifies the structure, the type and the meaning (through documentation) of the
-data.  With a schema, data can also be encoded more efficiently.
-
-If you don't know where to start, then we particularly recommend :ref:`Avro <serialization_and_evolution-avro>`, which
-is also supported in the Confluent Platform.  In the next section we briefly introduce Avro, and how it can be used to
-support typical schema evolution.
-
-
-.. _serialization_and_evolution-avro:
-
-Avro
-----
-
-
-Defining an Avro Schema
-^^^^^^^^^^^^^^^^^^^^^^^
-
-An Avro schema defines the data structure in a JSON format.
-
-The following is an example Avro schema that specifies a user record with two fields: ``name`` and ``favorite_number``
-of type ``string`` and ``int``, respectively.
-
-.. sourcecode:: json
-
-    {"namespace": "example.avro",
-     "type": "record",
-     "name": "user",
-     "fields": [
-         {"name": "name", "type": "string"},
-         {"name": "favorite_number",  "type": "int"}
-     ]
-    }
-
-You can then use this Avro schema, for example, to serialize a Java object (POJO) into bytes, and deserialize these
-bytes back into the Java object.
-
-One of the interesting things about Avro is that it not only requires
-a schema during data serialization, but also during data
-deserialization. Because the schema is provided at decoding time,
-metadata such as the field names don't have to be explicitly encoded
-in the data. This makes the binary encoding of Avro data very compact.
+.. _schema_evolution_and_compatibility:
 
+Schema Evolution and Compatibility
+==================================
 
 Schema Evolution
-^^^^^^^^^^^^^^^^
+----------------
 
 An important aspect of data management is **schema evolution**.  After the initial schema is defined, applications may
 need to evolve it over time. When this happens, it's critical for the downstream consumers to be able to handle data
 encoded with both the old and the new schema seamlessly. This is an area that tends to be overlooked in practice until
 you run into your first production issues.  Without thinking through data management and schema evolution carefully,
 people often pay a much higher cost later on.
 
-There are three common patterns of schema evolution:
+Note: when using Avro, one of the most important things is to manage its schemas and reason about how those
+schemas should evolve. :ref:`Confluent Schema Registry <schemaregistry_intro>` is built for exactly that purpose.
+You can find out the details on how to use it to store Avro schemas and enforce certain compatibility rules during
+schema evolution by looking at the :ref:`schemaregistry_api`.
+
```

Comment:
Maybe add something like this to explain about version: 

Schema compatibility is implemented in Schema Registry by versioning every single schema. When a schema is created the first time, it becomes version number 1 of the schema. When it is updated, the new updated schema becomes version 2 of the schema. Compatibility type determines how Schema Registry compares newer version of a schema against the older ones.

Reply:
Excellent!